nova/virt/vmwareapi/vmops.py

```diff
@@ -1499,6 +1499,9 @@ def _create_vm_clone(self, instance, vm_ref, snapshot_ref, dc_info,
         config_spec.numCoresPerSocket = 1
         config_spec.memoryMB = 16
         config_spec.uuid = image_id  # Not instanceUuid,
+        config_spec.extraConfig = vm_util.create_extra_config(client_factory,
+            {'nvp.vm-uuid': None})  # Another way to identify the vm
```

Comment:
Does `None` work? In pyVmomi I had to set it to `''`.

Reply:
We also set it to `''` in the volume-detach case: https://github.com/sapcc/nova/blob/fb37107cb1352043c6d7644d65dc94e8e8b1aed6/nova/virt/vmwareapi/volumeops.py#L122-L123

Reply:
True, it needs to be `''`, `None` raises an exception. I've changed it.